src/api/Device.api.ts

```diff
@@ -11,15 +11,28 @@ export default {
   getLatestEvents,
 };
 
+export interface DeviceInfo {
+  deviceName: string;
+  groupName: string;
+  id: number;
+  users?: UserDetails;
+}
+
+export interface UserDetails {
+  userName: string;
+  id: number;
+  admin: boolean;
+}
+
 function getDevices() {
   return CacophonyApi.get(
     `/api/v1/devices${shouldViewAsSuperUser() ? "" : "?view-mode=user"}`
   );
 }
 
-function getDevice(deviceNameOrId: string | number) {
+function getDevice(groupName: string, deviceName: string): Promise<DeviceInfo> {
```

Comment:
I don't think the return type in your type annotation is correct, it's going to be something like `{ result: {device: DeviceInfo }, ... }` right?  Unless you want to do the unwrapping for that in this API call?